src/transformers/models/llama/modeling_llama.py

```diff
@@ -278,9 +279,10 @@ def repeat_kv(hidden_states: torch.Tensor, n_rep: int) -> torch.Tensor:
 class LlamaAttention(nn.Module):
     """Multi-headed attention from 'Attention Is All You Need' paper"""
 
-    def __init__(self, config: LlamaConfig):
+    def __init__(self, config: LlamaConfig, layer_idx: int):
         super().__init__()
         self.config = config
+        self.layer_idx = layer_idx
```

Comment:
```suggestion
        self.layer_idx = layer_idx
        
        if layer_idx is None:
            logger.warn("Instantiating {self.__class__.__name__} without passing `layer_idx` is not recommended and can lead to errors during the forward call. Please make sure to provide a `layer_idx` when creating this class.")
```

Reply:
Added `warning_once` instead, to avoid a giant list of repeated warnings :)